apps/cow-tools/src/hooks/useUserMilkmanTransactions.ts

```diff
@@ -0,0 +1,171 @@
+import { Address } from "@bleu-fi/utils";
+import { useSafeAppsSDK } from "@gnosis.pm/safe-apps-react-sdk";
+import { gql } from "graphql-tag";
+import { useEffect, useState } from "react";
+import { createPublicClient, http, PublicClient } from "viem";
+import { goerli } from "viem/chains";
+
+import { erc20Abi } from "#/lib/abis/erc20";
+import { AllTransactionFromUserQuery } from "#/lib/gql/generated";
+import { sdk } from "#/lib/gql/sdk";
+
+gql(`
+  query AllTransactionFromUser ($user: String!) {
+    users(
+      where : { id : $user}
+    ) {
+      id
+      transactions {
+        id
+        blockNumber
+        blockTimestamp
+        swaps {
+          id
+          chainId
+          transactionHash
+          tokenAmountIn
+          priceChecker
+          orderContract
+          priceCheckerData
+          tokenIn {
+            id
+            name
+            symbol
+            decimals
+          }
+          tokenOut {
+            id
+            name
+            symbol
+            decimals
+          }
+        }
+      }
+    }
+  }
+`);
+
+export interface ICowOrder {
+  sellToken: Address;
+  buyToken: Address;
+  receiver: Address;
+  sellAmount: number;
+  buyAmount: number;
+  validTo: number;
+  feeAmount: number;
+  kind: string;
+  partiallyFillable: boolean;
+  sellTokenBalance: string;
+  buyTokenBalance: string;
+  from: Address;
+  executedSellAmount?: number;
+  executedSellAmountBeforeFees?: number;
+  executedBuyAmount?: number;
+  executedFeeAmount?: number;
+  status: string;
+}
+
+const cowApiUrl = "https://api.cow.fi/goerli";
+
+export function useUserMilkmanTransactions() {
+  const { safe } = useSafeAppsSDK();
+  const [loaded, setLoaded] = useState(false);
+  const [transactions, setTransactions] =
+    useState<AllTransactionFromUserQuery["users"][0]["transactions"]>();
+  const [cowOrders, setOrders] = useState<ICowOrder[][][]>();
+  const [hasToken, setHasToken] = useState<boolean[][]>();
+
+  const publicClient = createPublicClient({
+    chain: goerli,
+    transport: http(),
+  });
+
+  useEffect(() => {
+    async function loadOrders() {
+      const { users } = await sdk.AllTransactionFromUser({
+        user: safe.safeAddress,
+      });
```

Comment:
this is getting all transactions from safe, not only from milkman, right?

Reply:
Actually, the `sdk` is our ponder api. So it is getting, just the milkman.